ray_lightning/ray_ddp.py

```diff
@@ -553,6 +218,23 @@ def root_device(self):
         else:
             return torch.device("cpu")
 
+    @root_device.setter
+    def root_device(self, device):
+        self._device = device
+
+    def set_cuda_device_if_used(self):
```

Comment:
Same exact code as in `HorovodRayStrategy`. Should we extract this common method to `_GPUAccelerator`?

Reply:
```python
        if isinstance(trainer.strategy.accelerator, _GPUAccelerator): 
            trainer.strategy.accelerator.set_cuda_device_if_used(trainer.strategy)
```

Reply:
i figured out a better approach for this, maybe just as a utils function.

Reply:
then, there is no need to make sure whether the accelerator is the `_gpu`

Reply:
done